Code/GraphMol/FileParsers/PDBParser.cpp

```diff
@@ -296,6 +296,23 @@ static void PDBBondLine(RWMol *mol, const char *ptr, unsigned int len,
       } catch (boost::bad_lexical_cast &) {
         fail = true;
       }
+      // do not make bonds to metals
+      unsigned int elem = amap[src]->getAtomicNum();
```

Comment:
Yes, we can also include halogen anions as well. Although these would require further checkup (e.g. if the residue number is different).

Reply:
I will move this code out to a separate function, although both blocks serve different purpose. The one in `PDBBondLine()` avoids adding non-covalent bonds, which usually are present in PDB files (the official ones from wwPDB). The other one, in `IsBonded()`, avoidscreating bonds by the RDKit's `ConnectTheDots()`, which connects every pair of atoms which is closer than the sum of covalent radii. I will add `IsBlacklistedPair(Atom * beg_atom, Atom *end_atom)`, which will stop `ConnectTheDots()` making bonds to:
* metals
* halogen anions
* noble gasses
But only if they are in separate residue. This will also allow for metalorganic compounds in HETATM residues. I could also inject `HOH` unbinding here.